### PR TITLE
Return empty state if the Helm Release has not been installed

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#1902

### CODE REVIEW

File: service/chartconfig/v1/resource/chart/current.go [hunk 1]
```diff
@@ -15,12 +16,18 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	}
 
 	releaseContent, err := r.helmClient.GetReleaseContent(customObject)
-	if err != nil {
+	if helm.IsReleaseNotFound(err) {
+		// Fall through.
+		return &ChartState{}, nil
+	} else if err != nil {
```
Comment (rossf7) [inline service/chartconfig/v1/resource/chart/current.go]: We need to fall through if the release has not yet been installed.

File: service/chartconfig/v1/resource/chart/current.go [hunk 2]
```diff
@@ -15,12 +16,18 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	}
 
 	releaseContent, err := r.helmClient.GetReleaseContent(customObject)
-	if err != nil {
+	if helm.IsReleaseNotFound(err) {
+		// Fall through.
+		return &ChartState{}, nil
```
Comment (xh3b4sd) [inline service/chartconfig/v1/resource/chart/current.go]: This is not falling through. This is returning early. And why not `nil`?

Comment (rossf7) [reply service/chartconfig/v1/resource/chart/current.go]: Ack, good point it is returning early. I updated the comment and it now returns nil.

File: service/chartconfig/v1/resource/chart/current.go [hunk 3]
```diff
@@ -15,7 +16,10 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	}
 
 	releaseContent, err := r.helmClient.GetReleaseContent(customObject)
-	if err != nil {
+	if helm.IsReleaseNotFound(err) {
+		// Return early as release is not installed.
+		return nil, nil
+	} else if err != nil {
```
Comment (rossf7) [inline service/chartconfig/v1/resource/chart/current.go]: Here is the new error handling. Going with this now.

**Pull request merged**